File: C++/2593. Find Score of an Array After Marking All Elements.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long findScore(vector<int>& nums)
    {
        vector<pair<int, int>>vec;
        for (int i = 0;i < nums.size();++i)
            vec.push_back({ nums[i],i });

        sort(vec.begin(), vec.end());

        long long ans = 0;

        for (auto it : vec)
        {
            if (nums[it.second] > 0)
            {
                ans += nums[it.second];
                nums[it.second] *= -1;
                if (it.second - 1 >= 0)
                    nums[it.second - 1] = abs(nums[it.second - 1]) * -1;
                if (it.second + 1 < nums.size())
                    nums[it.second + 1] = abs(nums[it.second + 1]) * -1;
            }
        }
        return ans;
    }
};

File: Python/2593. Find Score of an Array After Marking All Elements.py
from typing import List


class Solution:
    def findScore(self, nums: List[int]) -> int:
        n = len(nums)
        vec = [(nums[i], i) for i in range(n)]
        vec.sort()

        ans = 0

        for value, index in vec:
            if nums[index] > 0:
                ans += nums[index]
                nums[index] *= -1
                if index - 1 >= 0:
                    nums[index - 1] = abs(nums[index - 1]) * -1
                if index + 1 < n:
                    nums[index + 1] = abs(nums[index + 1]) * -1

        return ans
